core/api/serializers/project_history.py

```diff
@@ -0,0 +1,56 @@
+from rest_framework import serializers
+
+from django.contrib.auth import get_user_model
+
+from core.models.project_history import ProjectHistory
+from core.models.project import Project
+
+User = get_user_model()
+
+
+class ProjectHistorySerializer(serializers.ModelSerializer):
+
+    project_id = serializers.PrimaryKeyRelatedField(
+        required=False,
+        queryset=Project.objects.all().values_list("id", flat=True),
+        write_only=True,
+    )
+
+    updated_by_id = serializers.PrimaryKeyRelatedField(
```

Comment:
Better to define a `UserSerializer` with fields = `["id", "username", "email", "first_name", "last_name"]`.

User in `ProjectHistorySerializer` as:
`user = UserSerializer()`

 On create, you pass `user = request.user` to the serializer